Fix thumbnail generation for command line

mschop opened the pull request:
When thumbnails are generated with -s flag, the thumbnail files are generated and directly removed, because the MediaDeletionSubscriber does delete the thumbnail files asynchronous via queue.

By setting the flag in the context, we can prevent that the deletion is done asynchronously.

<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://developer.shopware.com/docs/resources/guidelines/code/contribution?category=shopware-platform-dev-en/contribution).

Do your changes need to be mentioned in the documentation?
Please create a second pull request at https://github.com/shopware/docs
-->

### 1. Why is this change necessary?
It fixes the bug, that thumbnails are directly deleted after generation if invoked via command line `bin/console media:generate-thumbnails -s`.

### 2. What does this change do, exactly?
See above.

### 3. Describe each step to reproduce the issue or behaviour.

1. Delete all thumbnails.
2. Run `bin/console media:generate-thumbnails -s`

### 4. Please link to the relevant issues (if any).
None. I didn't create one yet.

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [x] I have written or adjusted the documentation according to my changes -> none required / bugfix
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

mschop committed:
Fix thumbnail generation for command line

When thumbnails are generated with -s flag, the thumbnail files are
generated and directly removed, because the MediaDeletionSubscriber
does delete the thumbnail files asynchronous via queue.

By setting the flag in the context, we can prevent that the deletion
is done asynchronously.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/shopware?pullRequest=3748) <br/>All committers have signed the CLA.

marcelbrode commented on src/Core/Content/Media/Commands/GenerateThumbnailsCommand.php:
```diff
@@ -199,6 +200,7 @@ private function createCriteria(): Criteria
      */
     private function generateSynchronous(RepositoryIterator $mediaIterator, Context $context): void
     {
+	$context->addState(MediaDeletionSubscriber::SYNCHRONE_FILE_DELETE);
```
Please indent

mschop replied:
Done

marcelbrode commented on src/Core/Content/Media/Message/GenerateThumbnailsHandler.php:
```diff
@@ -29,10 +30,11 @@ public function __construct(
         $criteria = new Criteria();
         $criteria->addAssociation('mediaFolder.configuration.mediaThumbnailSizes');
-        $criteria->addFilter(new EqualsAnyFilter('media.id', $msg->getMediaIds()));
+	    $criteria->addFilter(new EqualsAnyFilter('media.id', $msg->getMediaIds()));
```
Please keep empty spaces instead of tabs

mschop replied:
Done

marcelbrode commented on changelog/_unreleased/2024-06-05-fix-thumbnail-generation-via-command-line.md:
```diff
@@ -0,0 +1,10 @@
+author_github: mschop
+---
+# Core
+* Fix thumbnail generation via command line for -s flag
```
```suggestion
* Fix thumbnail generation via command line for -s flag
```
Please always start your changelog with "Added", "Remove", "Changed" or "Deprecated"

mschop replied:
Done

marcelbrode reviewed (changes requested):
Beside those minor issues, this looks fine to me, I'll ask a more backend focus colleague to have a look at it as well.

In any case, please add some tests for your change :)

marcelbrode commented:
Hi there! Any updates on this ticket?

marcelbrode commented:
Since we noticed no changes in the last 4 weeks, I'll close this ticket. Feel free to ping me, if you've rebased your branch and provided the changes asked and I will gladly reopen this ticket. Thank you :smile:

mschop commented:
@marcelbrode

Sorry, that it took so long. I am running very low on time.

I changed the things you mentioned and rebased the branch.

I currently don't have the time yet to add proper automated tests for this functionality. Could some developer on your side add that?

mitelg commented:
would you mind recreating the PR? I could not reopen it here :thinking:

mschop commented:
@mitelg Done: https://github.com/shopware/shopware/pull/3792
